sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -513,4 +541,23 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
       spark.catalog.uncacheTable("t2")
     }
   }
+
+  test("[SPARK-15870] DataFrame can't execute after uncacheTable") {
+    val selectStar = sql("SELECT * FROM testData WHERE key = 1")
+    selectStar.createOrReplaceTempView("selectStar")
+
+    spark.catalog.cacheTable("selectStar")
+    assert(
```

Comment:
this assert is unnecessary, we are testing the behaviour of caching and uncaching the same DataFrame, not validate the correctness of caching.